src/controllers/IndexController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Scout plugin for Craft CMS 3.x.
+ *
+ * Craft Scout provides a simple solution for adding full-text search to your entries. Scout will automatically keep your search indexes in sync with your entries.
+ *
+ * @link      https://rias.be
+ *
+ * @copyright Copyright (c) 2017 Rias
+ */
+
+namespace rias\scout\controllers;
+
+use Craft;
+use craft\web\Controller;
+use rias\scout\models\AlgoliaIndex;
+use rias\scout\Scout;
+use yii\web\Response;
+
+class IndexController extends Controller
+{
+    // Public Methods
+    // =========================================================================
+
+    /**
+     * Flush one or all Algolia indexes.
+     *
+     * @return Response
+     */
+    public function actionFlush()
+    {
+        $this->requirePostRequest();
+
+        $request = Craft::$app->getRequest();
+
+        $index = $request->getBodyParam('flushIndex');
+
+        try {
+            $mappings = Scout::$plugin->scoutService->getMappings($index);
+            $indexCount = count($mappings);
+
+            /* @var AlgoliaIndex $mapping */
+            foreach ($mappings as $mapping) {
+                $index = Scout::$plugin->scoutService->getClient()->initIndex($mapping->indexName);
+                $index->clearObjects();
+            }
+
+            Craft::$app->getSession()->setNotice(
+                Craft::t('scout', 'Flushed {indexCount} {indexNoun}.', [
+                    'indexCount' => $indexCount,
+                    'indexNoun'  => $indexCount === 1 ? 'index' : 'indices',
+                ])
+            );
+        } catch (\Throwable $e) {
+            Craft::$app->getSession()->setError($e->getMessage());
+        }
+
+        return $this->redirect(Craft::$app->getRequest()->getReferrer());
+    }
```

Comment:
Note that this flushing action currently runs without a confirmation step. I'm not sure if that is cause for concern or not, but thought I'd raise it in case it was.